tst/EnergyPlus/unit/ScheduleManager.unit.cc

```diff
@@ -804,3 +806,445 @@ TEST_F(EnergyPlusFixture, Schedule_GetCurrentScheduleValue_DST)
+    WeatherManager::WFAllowsLeapYears = false;
+    WeatherManager::LeapYearAdd = 0;
+
+    // ScheduleManager always assume LeapYear really.
+    int nDays = 365;
+    int NumOfTimeStepInHour = 4;
+
+    ScheduleManager::ScheduleInputProcessed = true;
+    EXPECT_TRUE(ScheduleManager::ScheduleInputProcessed);
+    ScheduleManager::NumSchedules = 1;
+    ScheduleManager::Schedule.allocate(ScheduleManager::NumSchedules);
+
+    ScheduleManager::Schedule(1).WeekSchedulePointer.allocate(366);
+    ScheduleManager::Schedule(1).WeekSchedulePointer = -1;
+    ScheduleManager::WeekSchedule.allocate(366);
+    ScheduleManager::DaySchedule.allocate(nDays); // Here only creating 365 ScheduleDays
+
+    Array1D_int EndDayOfMonth(12, {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31});
+
+    int dayOfYear = 0;
+    for (int month=1; month <= 12; ++month) {
+        for (int day = 1; day <= EndDayOfMonth(month); ++day) {
+            ++dayOfYear;
+            int DayOfYear_Schedule = General::OrdinalDay(month, day, 1);
+            if (month <= 2) {
+                EXPECT_EQ(dayOfYear, DayOfYear_Schedule);
+            } else {
+                EXPECT_EQ(dayOfYear+1, DayOfYear_Schedule);
+            }
+
+            ScheduleManager::Schedule(1).WeekSchedulePointer(DayOfYear_Schedule) = DayOfYear_Schedule;
+            ScheduleManager::WeekSchedule(DayOfYear_Schedule).DaySchedulePointer.allocate(7);
+            for (int d = 1; d <= 7; ++d) {
+                ScheduleManager::WeekSchedule(DayOfYear_Schedule).DaySchedulePointer(d) = dayOfYear;
+            }
+            ScheduleManager::DaySchedule(dayOfYear).TSValue.allocate(4, 24);
+            for (int whichHour = 1; whichHour <= 24; whichHour++) {
+                for (int TS = 1; TS <= NumOfTimeStepInHour; ++TS) {
+                    ScheduleManager::DaySchedule(dayOfYear).TSValue(TS, whichHour) = whichHour + (dayOfYear-1) * 24;
+                }
+            }
+        }
+    }
+
+    // Feb 28
+    EXPECT_EQ(59, ScheduleManager::Schedule(1).WeekSchedulePointer(59));
+    EXPECT_EQ(59, ScheduleManager::WeekSchedule(59).DaySchedulePointer(1));
+    EXPECT_EQ(59*24.0, ScheduleManager::DaySchedule(59).TSValue(4, 24));
+
+    // Feb 29: doesn't exist, and I default initialized everything above to -1
+    EXPECT_EQ(-1, ScheduleManager::Schedule(1).WeekSchedulePointer(60));
+    // ProcessSchedule would have treated the "Until: 3/1" to include the 2/29, so do that too.
+    ScheduleManager::Schedule(1).WeekSchedulePointer(60) = 61;
+
+    // March 1
+    EXPECT_EQ(61, ScheduleManager::Schedule(1).WeekSchedulePointer(61));
+    EXPECT_EQ(60, ScheduleManager::WeekSchedule(61).DaySchedulePointer(1));
+    EXPECT_EQ(60*24.0, ScheduleManager::DaySchedule(60).TSValue(4, 24));
+
+
+    EXPECT_EQ(366, ScheduleManager::Schedule(1).WeekSchedulePointer(366));
+    EXPECT_EQ(365, ScheduleManager::WeekSchedule(366).DaySchedulePointer(1));
+    EXPECT_EQ(8760.0, ScheduleManager::DaySchedule(365).TSValue(4, 24));
```

Comment:
Start by checking feb 28, Feb 29 (ordinal day 60, unused here), and the end of year value.